stablehlo/reference/Ops.cpp

```diff
@@ -68,6 +81,50 @@ Tensor eval(FloorOp op, const Tensor &operand) {
   return result;
 }
 
+Tensor eval(IotaOp op) {
+  Tensor result(op.getType());
+  ArrayRef<int64_t> shape = result.getType().getShape();
+  Type type = result.getType().getElementType();
+  if (!isSupportedIntegerType(type) && !isSupportedFloatType(type) &&
```

Comment:
This check can be removed. Instead, for every type cases enumerated below, return the `result` just within the `if` conditional and at the end of the function report_fatal_error.

Reply:
Will keep this per offline discussion to terminate early upon encountering error. This also removes the need for reporting error in the loop.